Search/ObjectToDocumentConverter.php

```diff
@@ -188,7 +188,13 @@ private function populateDocument($document, $object, $fieldMapping, $prefix = '
             if ($type !== Field::TYPE_STRING && $type !== Field::TYPE_ARRAY) {
                 $value = $this->converterManager->convert($value, $type);
 
-                $type = is_array($value) ? Field::TYPE_ARRAY : Field::TYPE_STRING;
+                $type = (is_null($value) ?
+                    Field::TYPE_NULL :
+                    (is_array($value) ?
+                        Field::TYPE_ARRAY :
+                        Field::TYPE_STRING
```

Comment:
And when looking at it deeper, I would not nest ternary operators, can you rewrite that logic with ifs?